### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by chirag-ti**
PR for the assignment on logging

### CODE REVIEW

File: fast_api_als/routers/lead_conversion.py [hunk 1]
```diff
@@ -67,4 +70,5 @@ async def submit(file: Request, token: str = Depends(get_token)):
         }
     else:
         # throw proper HTTPException
-        pass
+        raise HTTPException(status_code=404 , detail="Failed to update lead conversion status")
+        return
```
Comment (naman-TI) [inline fast_api_als/routers/lead_conversion.py]: Program halts once exception is raised hence 'return' is not needed.

File: fast_api_als/routers/three_pl.py [hunk 2]
```diff
@@ -17,9 +17,11 @@ async def reset_authkey(request: Request, token: str = Depends(get_token)):
     body = json.loads(body)
     provider, role = get_user_role(token)
     if role != "ADMIN" and (role != "3PL"):
+        logging.error("Invalid user")
```
Comment (naman-TI) [inline fast_api_als/routers/three_pl.py]: raise http exception with proper status code

File: fast_api_als/services/enrich_lead.py [hunk 3]
```diff
@@ -12,4 +12,7 @@
 
 
 def get_enriched_lead_json(adf_json: dict) -> dict:
+    if adf_json is not dict:
+        raise ValueError(f'given paramerter {adf_json} is in invalid format')
+        return
     pass
```
Comment (naman-TI) [inline fast_api_als/services/enrich_lead.py]: check for KeyError as well

**Review by naman-TI — CHANGES_REQUESTED** [review]
Exception handling is missing in many places and some files have not been looked at. Not Approved.

File: fast_api_als/routers/three_pl.py (hunk 2 above)
Comment (chirag-ti) [reply fast_api_als/routers/three_pl.py]: Resolved

File: fast_api_als/services/enrich_lead.py (hunk 3 above)
Comment (chirag-ti) [reply fast_api_als/services/enrich_lead.py]: Added Key Error case

File: fast_api_als/routers/three_pl.py (hunk 2 above)
Comment (chirag-ti) [reply fast_api_als/routers/three_pl.py]: Improved the http exceptions with proper status codes in all files, please check now

**Comment by naman-TI** [comment]
use `logger = logging.getLogger(__name__)` in each module. Loggers should never be instantiated directly, but always through the module-level function.

**Pull request merged**